compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -12,20 +13,28 @@ module DA.Daml.LF.Proto3.DecodeV1
 import           DA.Daml.LF.Ast as LF
 import           DA.Daml.LF.Proto3.Error
 import Control.Monad
+import Control.Monad.Error.Class (MonadError(throwError))
+import Control.Monad.Reader (ReaderT, ask, runReaderT)
 import Text.Read
 import           Data.List
 import           DA.Daml.LF.Mangling
 import qualified Da.DamlLf1 as LF1
-import           Data.Either.Combinators (mapLeft)
 import qualified Data.NameMap as NM
 import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 import qualified Data.Vector as V
 import qualified Proto3.Suite as Proto
 
+-- internal functions that *implement* decoding
+type DecodeImpl = ReaderT PackageRefCtx Decode
+type MDecode = MonadError Error
```

Comment:
Can we please spell this out as `MonadDecode`?

Reply:
Sure. a7373a5dc882c784862c0d8f44702bb925005776